wellcomeml/ml/keras_vectorizer.py

```diff
@@ -21,20 +24,59 @@ def transform(self, X, *_):
         sequences = self.tokenizer.texts_to_sequences(X)
         return pad_sequences(sequences, maxlen=self.sequence_length)
 
-    def build_embedding_matrix(self, embeddings_path):
-        embeddings_index = {}
-        with open(embeddings_path) as f:
-            for line in f:
-                word, coefs = line.split(maxsplit=1)
-                coefs = np.fromstring(coefs, "f", sep=" ")
-                embeddings_index[word] = coefs
+    def build_embedding_matrix(self, embeddings_name_or_path=None):
+        """
+        Builds an embedding matrix from either a local embeddings path
+        or a gensim pre-trained word vector path
+
+        Args:
+            embeddings_name_or_path:
+                Can be either:
+                - A local directory to word embeddings
+                - The name of a GenSim pre-trained word vector model
+                    e.g. 'glove-twitter-25', for the complete list:
+                    https://github.com/RaRe-Technologies/gensim-data#models
+
+        Returns:
+            An embedding matrix
+
+        """
+        local_embeddings = False
+        if path.isfile(embeddings_name_or_path):
+            try:
+                embeddings_index = {}
+                with open(embeddings_name_or_path) as f:
+                    for line in f:
+                        word, coefs = line.split(maxsplit=1)
+                        coefs = np.fromstring(coefs, "f", sep=" ")
+                        embeddings_index[word] = coefs
+                    emb_dim = len(coefs)
+                local_embeddings = True
+            except TypeError:
+                raise TypeError("Incorrect local embeddings path")
+        elif embeddings_name_or_path:
+            try:
+                embeddings_index = api.load(embeddings_name_or_path)
+                emb_dim = embeddings_index.vector_size
+            except ValueError:
+                raise ValueError(
+                    "Incorrect GenSim word vector model name, try e.g. 'glove-twitter-25'"
+                )
+        else:
+            raise TypeError("No local or GenSim word embeddings given")
 
-        emb_dim = len(coefs)
         num_words = len(self.tokenizer.word_index) + 1
 
         embedding_matrix = np.zeros((num_words, emb_dim))
         for word, i in self.tokenizer.word_index.items():
-            embedding_vector = embeddings_index.get(word)
+            if local_embeddings:
```

Comment:
is that needed, why not use the try except block for both cases?

Reply:
I think it is because the `get` command wont work for the gensim embeddings case

Reply:
I think it is because the `get` command wont work for the gensim embeddings case